src/movies.js

```diff
@@ -1,22 +1,72 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+  return moviesArray.map(function (element) {
+    return element.director;
+  });
+}
+
+const directors = getAllDirectors(movies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
```

Comment:
First whe need to check the lenght of the movies array 
```suggestion
function howManyMovies(moviesArray) {
  if (!moviesArray.length) {
        return 0
    }
```